crypto/x509/x509_att.c

```diff
@@ -322,6 +322,9 @@ int X509_ATTRIBUTE_set1_data(X509_ATTRIBUTE *attr, int attrtype,
         }
     } else {
         ASN1_TYPE_set(ttmp, atype, stmp);
+        if (atype == V_ASN1_BOOLEAN) {
```

Comment:
Somehow I think a boolean should be an error here, hey should go thru
the ` if (len == -1 && (attrtype & MBSTRING_FLAG) == 0) {` above.

Reply:
Well, `MSBSTRING_FLAG` won't be set but IMO it is possible to call it with `len != -1`. So it seems to me it is possible to get here with V_ASN1_BOOLEAN.

Reply:
yes, but it is impossible to encode a `FALSE`, even when `data == NULL` and `len == 0`,
`stmp` will always be non-zero.

Reply:
Yes, and also `V_ASN1_NULL` will probably also result in a memory leak here, because it stores the pointer,
but the pointer should be `NULL` because it will not be freed by `asn1_primitive_free`.